scripts/apps/authoring/authoring/index.ts

```diff
@@ -216,8 +216,8 @@ angular.module('superdesk.apps.authoring', [
                     {action: 'list', type: 'archive'},
                     {action: 'edit', type: 'item'},
                 ],
-                additionalCondition: ['authoring', 'item', function(authoring, item) {
-                    return authoring.itemActions(item).edit;
+                additionalCondition: ['authoring', 'item', async function(authoring, item) {
```

Comment:
For one we aren't using async/await in the codebase, so you'd need to rewrite this using promises. On the other hand, `itemActions` function is synchronous so neither `.then` nor `await` can be used.

Reply:
you should probably step into use a debugger to step into `authoringWorkspace.edit` on line 213, go to the implementation and do the thing you tried to do in the initial PR version. I'm still not sure actually what's the issue and how you are trying to fix it. Can you comment on that too?

Reply:
When we are trying to edit locked packages inside locked packages then We are not getting response on time from 
return `authoring.itemActions(item)` and because of that error which is not handled by us,and we are not able to reach at 
`authoringWorkspace.edit(data.item ? data.item : data);`  so, the main problem is not in this , problem lies in `return authoring.itemActions(item).`


![image (12)](https://user-images.githubusercontent.com/73937490/210788577-96d37b01-784c-4d4e-843b-82c0f9d935e9.png)

Reply:
> We are not getting response on time

Which line in your opinion is asynchronous? I looked into it and it seems to me that we do get the response, it's simply `false` - I mean it refused to allow to edit it. I've looked into removing the condition in the first place, but it would likely change behavior somewhere else so I applied a targeted fix in packages.